base/linalg/dense.jl

```diff
@@ -18,6 +20,8 @@ function scale!{T<:BlasFloat}(X::Array{T}, s::T)
     X
 end
 
+scale!{T<:BlasFloat}(s::T, X::Array{T}) = scale!(X, s)
```

Comment:
why does this method exist? really strange to be sometimes mutating the second input, but usually the first